### PR TITLE
add accountId to managed policies

### PR DESCRIPTION
**Pull request by tylersouthwick**

### CODE REVIEW

File: src/main/scala/com/monsanto/arch/cloudformation/model/resource/IAM.scala [hunk 1]
```diff
@@ -83,9 +83,8 @@ object `AWS::IAM::ManagedPolicy` extends DefaultJsonProtocol {
   implicit val format: JsonFormat[`AWS::IAM::ManagedPolicy`] = jsonFormat8(`AWS::IAM::ManagedPolicy`.apply)
 }
 
-
-case class AWSManagedPolicy(name: String) {
-  def buildARN = s"arn:aws:iam::aws:policy/$name"
+case class AWSManagedPolicy(name: Token[String], accountId : Token[String] = "aws") {
```
Comment (tj-corrigan) [inline src/main/scala/com/monsanto/arch/cloudformation/model/resource/IAM.scala]: I'm curious when the accountId here would ever be something besides aws? Are you using this as a workaround to not having a ResourceRef of a policy?

**Comment by tylersouthwick** [comment]
ping

**Comment by bkrodgers** [comment]
Hi Tyler, did you see TJ's comment on the PR earlier?  Was looking for an answer before I merged (though it's not really an issue as is).

File: src/main/scala/com/monsanto/arch/cloudformation/model/resource/IAM.scala (hunk 1 above)
Comment (tylersouthwick) [inline src/main/scala/com/monsanto/arch/cloudformation/model/resource/IAM.scala]: Yes. This is when there is a policy created, not through cloudformation, but manually. The other approach I could see would be to create another case class and implicit converter and modify ManagedPolicyARN to have a third type (or have the different types extend a sealed trait). thoughts?

Comment (bkrodgers) [inline src/main/scala/com/monsanto/arch/cloudformation/model/resource/IAM.scala]: Just based on naming, I'd probably be inclined to say let's have a separate case class, called something like `ExternalManagedPolicy`, `ExistingManagedPolicy`, or whatever.  `AWSManagedPolicy` was intended to be for a "Managed Policy" that's managed by AWS.  It wouldn't be the end of the world to keep it as you've submitted it, but it'd be somewhat clearer to go the other way.

**Comment by tylersouthwick** [comment]
@bkrodgers  @tj-corrigan I re-worked it to add another case class instead. can you take a look?

**Comment by tylersouthwick** [comment]
ping

**Comment by bkrodgers** [comment]
Sorry for the delay, been really busy.  I'll see if I can find time to catch up today.  If not, I'll be out until Tues after today, but I'll make sure this gets done next week.

**Comment by bkrodgers** [comment]
Sorry for the delay.  I think this looks good.  Thanks!

**Comment by tylersouthwick** [comment]
@bkrodgers can you release this change? Thanks!

**Comment by bkrodgers** [comment]
Yes, I had another PR I was waiting to see if I got an answer on, but I'll cut a release with this one today or tomorrow if you need it right away.

**Comment by bkrodgers** [comment]
3.6.1 released!

**Pull request merged**